File: leetcode57/solveLeetcode57.cpp
// //use leetcode56
// class Solution 
// {

//     vector<vector<int>> merge(vector<vector<int>>& intervals) 
//     {
//         const int n=intervals.size(),maxn=1e4+7;
//         sort(intervals.begin(),intervals.end());

//         vector<vector<int>>ans;

//         int i;
//         for(i=0;i<n;++i)
//         {
//             if(ans.empty() || ans.back()[1]<intervals[i][0])
//                 ans.emplace_back(intervals[i]);
//             else ans.back()[1]=max(intervals[i][1],ans.back()[1]);
//         }

//         return ans;
//     }
// public:
//     vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) 
//     {
//         intervals.emplace_back(newInterval);
//         return merge(intervals);
//     }
// };

File: leetcode57/solveLeetcode57.go
// func insert(intervals [][]int, newInterval []int) [][]int {
// 	res := make([][]int, 0)
// 	l := len(intervals)
// 	i := 0
// 	for i < l && intervals[i][1] < newInterval[0] {
// 		res = append(res, intervals[i])
// 		i++
// 	}
// 	for i < l && intervals[i][0] <= newInterval[1] {
// 		newInterval[0] = min(newInterval[0], intervals[i][0])
// 		newInterval[1] = max(newInterval[1], intervals[i][1])
// 		i++
// 	}
// 	res = append(res, newInterval)
// 	for i < l {
// 		res = append(res, intervals[i])
// 		i++
// 	}
// 	return res
// }

// func merge(intervals [][]int) [][]int {

// 	n := len(intervals)

// 	sort.Slice(intervals, func(i, j int) bool {

// 		return intervals[i][0] < intervals[j][0]

// 	})

// 	ans := [][]int{}

// 	for i := 0; i < n; i++ {

// 		if len(ans) == 0 || ans[len(ans)-1][1] < intervals[i][0] {

// 			ans = append(ans, intervals[i])

// 		} else {

// 			ans[len(ans)-1][1] = max(intervals[i][1], ans[len(ans)-1][1])

// 		}

// 	}

// 	return ans

// }

// func insert(intervals [][]int, newInterval []int) [][]int {

// 	intervals = append(intervals, newInterval)

// 	return merge(intervals)

// }